OpenBLAS 3.7.0 test fail when building without CBLAS and LAPACKE

When building OpenBLAS 3.7.0 with

```
make -j3 DYNAMIC_ARCH=1 FC=gfortran NO_CBLAS=1 NO_LAPACKE=1 USE_OPENMP=0
```

The following linker errors with missing LAPACKE and CBLAS functions are obtained.  This can be fully understood, as they are not built by intention.

```
/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: test_kernel_regress.o: in function `__ctest_kernel_regress_skx_avx_run':
test_kernel_regress.c:(.text+0xa7): undefined reference to `LAPACKE_dgesvd'
/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: test_kernel_regress.c:(.text+0x130): undefined reference to `cblas_dgemm'
/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: test_kernel_regress.c:(.text+0x171): undefined reference to `cblas_dnrm2'
```

This `test_kernel_regress` was introduced in a211bc9b6a6e597a38fc8b8b7ed0b006cb367c46 and should rather check for LAPACKE and CBLAS to be build, not only LAPACK.

Thank you for your help.

# System info

- openSUSE Leap 15.1
- Intel(R) Core(TM) i5-6200U CPU @ 2.30GHz

```
$ uname -a
Linux linux-p86y 4.12.14-lp151.28.10-default #1 SMP Sat Jul 13 17:59:31 UTC 2019 (0ab03b7) x86_64 x86_64 x86_64 GNU/Linux
```

Full build log: 
[OpenBLAS-0.3.7.build.log.tar.gz](https://github.com/xianyi/OpenBLAS/files/3495081/OpenBLAS-0.3.7.build.log.tar.gz)

--- a/utest/CMakeLists.txt
+++ b/utest/CMakeLists.txt
@@ -38,8 +38,13 @@ if (NOT NO_LAPACK)
 set(OpenBLAS_utest_src
   ${OpenBLAS_utest_src}
   test_potrs.c
+  )
+if (NOT NO_CBLAS AND NOT NO_LAPACKE)
+set(OpenBLAS_utest_src
+  ${OpenBLAS_utest_src}
   test_kernel_regress.c
   )
+endif()
 endif()
 
 set(OpenBLAS_utest_bin openblas_utest)

--- a/utest/Makefile
+++ b/utest/Makefile
@@ -1,6 +1,9 @@
 UTEST_CHECK = 1
 TOPDIR	= ..
 
+override TARGET_ARCH=
+override TARGET_MACH=
+
 UTESTBIN=openblas_utest
 
 .PHONY : all
@@ -13,7 +16,11 @@ OBJS=utest_main.o test_amax.o test_rotmg.o test_axpy.o test_dotu.o test_dsdot.o 
 
 ifneq ($(NO_LAPACK), 1)
 OBJS += test_potrs.o
+ifneq ($(NO_CBLAS), 1)
+ifneq ($(NO_LAPACKE), 1)
 OBJS += test_kernel_regress.o
+endif
+endif
 endif
 
 #this does not work with OpenMP nor with native Windows or Android threads

